test/e2e/resize_nodes.go

```diff
@@ -38,38 +39,62 @@ import (
 const serveHostnameImage = "gcr.io/google_containers/serve_hostname:1.1"
 
 func resizeGroup(size int) error {
-	// TODO: make this hit the compute API directly instread of shelling out to gcloud.
-	output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--project="+testContext.CloudConfig.ProjectID, "--zone="+testContext.CloudConfig.Zone,
-		"resize", testContext.CloudConfig.NodeInstanceGroup, fmt.Sprintf("--new-size=%v", size)).CombinedOutput()
-	if err != nil {
-		Logf("Failed to resize node instance group: %v", string(output))
+	if testContext.Provider == "gce" {
+		// TODO: make this hit the compute API directly instread of shelling out to gcloud.
+		output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--project="+testContext.CloudConfig.ProjectID, "--zone="+testContext.CloudConfig.Zone,
+			"resize", testContext.CloudConfig.NodeInstanceGroup, fmt.Sprintf("--new-size=%v", size)).CombinedOutput()
+		if err != nil {
+			Logf("Failed to resize node instance group: %v", string(output))
+		}
+		return err
+	} else {
```

Comment:
Well the idea here is that most cloudproviders should just implement the interface (and we should pull it up out of cloudprovider/aws into cloudprovider).  We would eliminated the gcloud call and the provider-switching.  We could also check whether the provider implemented the interface rather than comparing vs a known list.

For now though, I'll add a comment to say "//Supported by aws" so we know which provider this is :-)